### PR TITLE
Welcome screen redesign

### PR DESCRIPTION
**Pull request by TheZ3ro**
## Description

Welcome Screen redesign

## Motivation and Context

A nicer welcome screen for new users

## How Has This Been Tested?

Manually on Linux (should work no problem on other platforms)

## Screenshots (if appropriate):
istantanea_2017-02-11_16-37-23

## Types of changes

- ✅ New feature (non-breaking change which adds functionality)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**

### CODE REVIEW

File: src/gui/WelcomeWidget.cpp [hunk 1]
```diff
@@ -18,13 +18,52 @@
 #include "WelcomeWidget.h"
 #include "ui_WelcomeWidget.h"
 
+#include <QDebug>
```
Comment (louib) [inline src/gui/WelcomeWidget.cpp]: @TheZ3ro I guess that was used for debugging while coding the feature?

**Comment by phoerious** [comment]
One more thing: You should either spell Database with a lowercase D or make everything title case.

**Comment by louib** [comment]
This is the UI on MacOs.

screen shot 2017-02-11 at 1 31 13 pm

The `Import from KeePass1` button is not the same size as the others.

Also, maybe we could hide the `Recent databases` if the list of recent databases is empty.

Good job @TheZ3ro !!

**Comment by phoerious** [comment]
I have the feeling that widget sizes on OS X are quite messed up in general. The keyfile dropdown in the unlock dialog is also not the same size as the password field. And the password fields in the new database widget are tiny. I think we need to polish all that a little for 2.2.0.

**Comment by TheZ3ro** [comment]
Oh, I know why the KeePass1 button is smaller. It's inside a horizontal layout with the (hide) CSV button.

File: src/gui/WelcomeWidget.ui [hunk 2]
```diff
@@ -2,17 +2,211 @@
...
+          </color>
+         </brush>
+        </colorrole>
+       </inactive>
+       <disabled>
+        <colorrole role="Button">
+         <brush brushstyle="SolidPattern">
+          <color alpha="255">
+           <red>0</red>
+           <green>255</green>
```
Comment (phoerious) [inline src/gui/WelcomeWidget.ui]: I would remove that color. The button should have the system's default button color.
The green is really lurid and also very surprising. When the button has focus, it has a nice light blue on my system, but when it loses focus, it's suddenly shocking green.

File: src/gui/WelcomeWidget.ui [hunk 3]
```diff
@@ -2,17 +2,211 @@
...
+      <string>Open existing database</string>
+     </property>
+    </widget>
+   </item>
+   <item>
+    <layout class="QHBoxLayout" name="horizontalLayout_2">
+     <item>
+      <widget class="QPushButton" name="buttonImportKeePass1">
+       <property name="text">
+        <string>Import from KeePass1</string>
```
Comment (phoerious) [inline src/gui/WelcomeWidget.ui]: The file filter in the open dialog has a space between "KeePass" and "1". I suggest adding one here, too.

File: src/gui/WelcomeWidget.ui (hunk 2 above)
Comment (TheZ3ro) [inline src/gui/WelcomeWidget.ui]: Seems that this wasn't reverted in the previus force-push

**Pull request merged**